comp/networkdeviceconfig/impl/networkdeviceconfig.go

```diff
@@ -27,24 +30,72 @@ type Provides struct {
 }
 
 type networkDeviceConfigImpl struct {
-	config config.Component
-	log    log.Component
+	config        *ProcessedNcmConfig
+	log           log.Component
+	clientFactory RemoteClientFactory
 }
 
 // NewComponent creates a new networkdeviceconfig component
 func NewComponent(reqs Requires) (Provides, error) {
+	ncmConfig := newConfig(reqs.Config)
```

Comment:
You are returning `nil` if there is an error, this will panic on line 54. 

I think instead of returning nil if something fails from `newConfig`, I would return `( *ProcessedNcmConfig, error)` and in the component would check for the error, and if there is some, I would `return Provides{}, err`

That way we warranty that when calling `RetrieveConfiguration`, things would not panic